### DISCUSSION TITLE
Dark Theme

### DISCUSSION THREAD

**Discussion by whitead** [executablebooks/meta#204] [Q&A]
Now that github has a nice dark theme, I'm thinking that a nice dark theme would be good for jupyter books. Any ideas how to implement this, aside from designing it myself in `_static` custom css. Can you swap the themes somehow?

**Comment by whitead** [comment]
I believe bootstrap is used, maybe something [like this](https://www.codeply.com/p/bV4XYd7H9h)? Do I fork the sphinx book theme for this or add to custom css/js? Thoughts?

**Comment by jstac** [comment]
These lectures are written with JB and have a dark theme -- see the icon on the bar up top: https://python-programming.quantecon.org/intro.html

That theme was created for QE but it's intended for sharing --- @drdrij is the creator.

**Reply by choldgraf** [reply]
more generally we'd also love to get dark theme support in the `sphinx-book-theme` as well!

**Reply by choldgraf** [reply]
btw @drdrij that is a beautiful theme I must say

**Reply by DrDrij** [reply]
Thanks @choldgraf :)

**Comment by DrDrij** [comment]
The method we've used for _quantecon_ is fairly straight forward. The initial build for the _quantecon_ theme is written with minimal markup and not based off bootstrap. A CSS class (in this case 'dark-theme') on the body element allows re-styling of any elements that need colour changes. I've used javascript to toggle the class and hence enabling the additional styles.

The example you've linked @whitead looks like a nice quick way to get a basic dark theme going. It will need an additional set of styles for custom elements. A potential neater way to do it would be leveraging CSS or SCSS variables (https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).

One experimental addition which we've added to the _quantecon_ theme is client-side inversion of image colours with CSS filters.

```
filter: invert(100%) hue-rotate(-180deg) !important;
-ms-filter: invert(100%) !important;
-webkit-filter: invert(100%) hue-rotate(-180deg) !important;
```

This works really well for graphs (eg. axis/plots on white background and console screenshots) but not for photos etc.
This may require some consideration for image-heavy books - is it necessary to create a light and dark version of images and lazy load the respective image when needed?

**Comment by chrisjsewell** [comment]
FYI, I just use https://github.com/darkreader/darkreader in Firefox, and so personally never really have the need for site specific dark modes.
But it might be worth looking at their code
